components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java

```diff
@@ -354,27 +354,25 @@ public static String formatDateTime(Date date) {
             return scimToLocalClaimMap;
         } catch (ClaimMetadataException e) {
             throw new UserStoreException("Error occurred while retrieving SCIM to Local claim mappings for tenant " +
-                    "domain : " + spTenantDomain, e);
+                    "domain : " + tenantDomain, e);
         }
     }
 
     /**
-     * This is used to get tenant domain of thread local service provider.
+     * This is used to get tenant domain.
      *
-     * @return Service provider's tenant domain.
+     * @return user's tenant domain.
      */
-    private static String getTenantDomainFromSP() {
+    private static String getTenantDomain() {
 
         String tenantDomain;
-        ThreadLocalProvisioningServiceProvider threadLocalSP = IdentityApplicationManagementUtil
-                .getThreadLocalProvisioningServiceProvider();
-        if (threadLocalSP != null) {
-            return threadLocalSP.getTenantDomain();
-        } else if (IdentityTenantUtil.getTenantDomainFromContext() != null) {
-            return IdentityTenantUtil.getTenantDomainFromContext();
-        } else if (PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain() != null) {
-            tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            tenantDomain = IdentityTenantUtil.getTenantDomainFromContext();
         } else {
+            tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        }
+
+        if (StringUtils.isBlank(tenantDomain)){
```

Comment:
Shall we add some debug log here? Saying tenant domain is empty, and hence reading it as super tenant

Reply:
Addressed commit with 20ad7091a8daf313ae8ae76c4ddd8e74cda25b3c